### PR TITLE
Add tests and run them in CI

### PR DESCRIPTION
**Pull request by antgamdia**
### Description

This pull request introduces comprehensive test coverage for the main modules and functions in the codebase, improves CI/CD automation, and updates configuration files for better consistency and reproducibility. The most important changes are summarized below:

* Added extensive unit and integration tests for the main functionalities in `src/config.rs`, `src/list.rs`, and `src/run.rs`, ensuring correct parsing, listing, and execution logic, including edge cases and error handling. 
* Added a new `lint` job and enhanced the `build` job in the GitHub Actions workflow (`.github/workflows/rust.yml`), including steps for version gathering, Rust toolchain setup, and Clippy linting to enforce code quality.

Related #TRNT-4628

### How was this tested?

CI

### Documentation changes

No

### Additional information

Note that before this PR we had 0 tests... so Dependabot's PR have been merged without proper test coverage...

### CODE REVIEW

File: src/run.rs [hunk 1]
```diff
@@ -159,3 +159,394 @@ fn extract_fixtures_from_directory(directory: &String) -> Option<Vec<String>> {
...
+        assert!(files[1].ends_with("b.json"));
+
+        fs::remove_dir_all(&dir).unwrap();
+    }
+
+    #[test]
+    fn scan_directory_errors_on_missing_directory() {
+        let dir = std::env::temp_dir().join("photofinish_test_scan_directory_missing");
+        assert!(scan_directory(dir.to_str().unwrap()).is_err());
+    }
```
[the remark this answers is not in the document]
Comment (antgamdia) [reply src/run.rs:251]: https://github.com/trento-project/photofinish/pull/57/commits/e6c45e226bbe1a9c0a3273756b856c2e786a5693

File: src/list.rs [hunk 2]
```diff
@@ -17,3 +17,34 @@ fn print_scenario(scenario: &Scenario) {
...
     )
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    // println! output isn't captured/asserted here (would need a refactor to inject
+    // a writer); these are smoke tests confirming the functions don't panic.
+
```
[the remark this answers is not in the document]
Comment (antgamdia) [reply src/list.rs:27]: https://github.com/trento-project/photofinish/pull/57/commits/e6c45e226bbe1a9c0a3273756b856c2e786a5693

File: .github/workflows/rust.yml [hunk 3]
```diff
@@ -5,24 +5,50 @@ name: Rust
...
+        uses: actions-rust-lang/setup-rust-toolchain@166cdcfd11aee3cb47222f9ddb555ce30ddb9659 # v1
+        with:
+          toolchain: ${{ env.RUST_VERSION }}
+      - name: Check license headers
+        uses: apache/skywalking-eyes/header@61275cc80d0798a405cb070f7d3a8aaf7cf2c2c1 # v0.8.0
+      - name: Install Clippy
+        run: rustup component add clippy
+      - name: Run Clippy
+        id: clippy
+        run: cargo clippy
```
[the remark this answers is not in the document]
Comment (antgamdia) [reply .github/workflows/rust.yml]: https://github.com/trento-project/photofinish/pull/57/commits/e6c45e226bbe1a9c0a3273756b856c2e786a5693

**Review by nelsonkopliku — APPROVED** [review]
